cmd/libsnap-confine-private/mount-opt.c

```diff
@@ -262,10 +265,9 @@ void sc_do_mount(const char *source, const char *target,
 #ifdef SNAP_CONFINE_DEBUG_BUILD
 		mount_cmd = sc_mount_cmd(buf, sizeof(buf), source,
```

Comment:
Maybe its just me, but I find that this code requires relatively much state in head of the reader. How about just keeping the structure with #ifdef/#else/#endif and just re-calcuating the mount_cmd when an error occurs (i.e. *removing* the if (mount_cmd == NULL) check below? If that is too confusing we could even not reuse the mount_cmd var in two places for different purposes but instead call one "debug_mount_cmd" or somesuch. I outlined what I have in mind in https://paste.ubuntu.com/26390503/ - I find the result easier to read linearly (less side-effects to keep in mind).